### ISSUE TITLE
When screen capturing, it is difficult to determine which monitor is which

### ISSUE THREAD

**Issue by garrett** [flatpak/xdg-desktop-portal#666]
I hope this is the right place to report the bug. If not, I'm happy to report elsewhere.

I have 2 external monitors. Both are Dell 27" displays. They're both the same resolution. In my case, they're different model numbers, but may people may buy two of the same monitor.

When I try to screen share in a video conference or use Kooha (screen recording app), I get a dialog that looks like this:

![Screenshot from 2021-11-29 17-46-30](https://user-images.githubusercontent.com/10246/143908611-c8d0c051-4f15-45c4-a2d4-0e00a237d2b9.png)

I have no way of knowing which 'Dell Inc. 27"' monitor is which! Is it my primary on the left or secondary on the right?

GNOME solves this in 2 ways:

1. The GNOME Settings app shows an overview map of the monitor layout and refer to the assigned number whenever the monitor is referenced.
2. At the same time, a number badge is overlaid on each monitor.

![Screenshot from 2021-11-29 17-48-28](https://user-images.githubusercontent.com/10246/143911854-135ed65d-828f-45a2-87ad-6bc6b71b2ed2.png)

Two different workarounds I could suggest:

1. Detect duplicate strings and get more specific
   1. If the descriptions are the same, show the model number as well.
   2. If the model numbers are the same, perhaps show some identifier of some sort.
2. If it's possible to get the orientation of the monitors, then it would be awesome to say something like:
   - 'Dell 27" Left'
   - 'Dell 27" Right'
   - (or 'Dell 27" Middle' otherwise)

---

So, to summarize, there are four different possible solutions:
1. Graphical map
2. Overlay
3. Showing model (or some other identifier) when there are otherwise duplicate strings
4. Showing spatial display order

**Comment by garrett** [comment]
I modified the screenshot and turned the monitors on the side, in the spacial order that I see them (going from left to right):

![share-monitor-horizontal](https://user-images.githubusercontent.com/10246/143911676-6458a950-75ab-4478-9782-846f11f7d3e4.png)

In this list, I'd know which is which, unlike the above. The portal could even show a border around the selected monitor, to reinforce which is selected.

**Comment by jadahl** [comment]
FWIW, the mockups include previews of the screen content. See https://gitlab.gnome.org/Teams/Design/os-mockups/-/blob/master/portals/src/screen-sharing-remote-desktop.svg. Anyhow, this issue should probably be moved to https://gitlab.gnome.org/GNOME/xdg-desktop-portal-gnome.
